packages/grid/src/vaadin-grid-selection-column-base-mixin.js

```diff
@@ -107,6 +107,7 @@ export const GridSelectionColumnBaseMixin = (superClass) =>
         root.appendChild(checkbox);
         // Add listener after appending, so we can skip the initial change event
         checkbox.addEventListener('checked-changed', this.__onSelectAllCheckedChanged.bind(this));
+        column._headerCell.addEventListener('keydown', this.__onHeaderCellKeydown.bind(this));
```

Comment:
Could we listen to `keydown` event on the grid element instead? Or could add this logic to KeyboardNavigationMixin? The current implementation can result in multiple listeners if the grid clears the cell content at some time.

Reply:
I don't think that using custom header renderer for `vaadin-grid-selection-column` is a supported case.
Moving code to the grid mixin would mean activating checkboxes in all grid cells on <kbd>Space</kbd>, should we do it?

Reply:
Tested by setting `hidden` property on the selection column and its header cell content doesn't get cleared.

Regarding adding a listener, we could move it to `_createCell()` method as it already has some listeners:
https://github.com/vaadin/web-components/blob/20e538d506beeebe814a22067834633e561bb941/packages/grid/src/vaadin-grid-mixin.js#L564

But again, this would imply that any header cell with `vaadin-checkbox` in it would allow to toggle it on <kbd>Space</kbd>.

Reply:
> Tested by setting hidden property on the selection column and its header cell content doesn't get cleared.

OK, thanks for checking 👍 

> Regarding adding a listener, we could move it to _createCell() method as it already has some listeners:
> ...
> But again, this would imply that any header cell with vaadin-checkbox in it would allow to toggle it on Space.

I was actually thinking that `_createCell()` could call a column method such as `_onCellKeyDown`, which could be overridden in the selection column to do this logic.

Reply:
I see. This would mean adding `keydown` listeners to all the cells though, which seems redundant to me.